cmd/jujud/agent/model/manifolds.go

```diff
@@ -159,23 +191,23 @@ func Manifolds(config ManifoldsConfig) dependency.Manifolds {
 		// fortress).
 		//
 		// Note that the fortress and flag will only exist while
-		// the model is not dead; this frees their dependencies
-		// from model-lifetime concerns.
-		migrationFortressName: ifNotDead(fortress.Manifold()),
-		migrationInactiveFlagName: ifNotDead(migrationflag.Manifold(migrationflag.ManifoldConfig{
+		// the model is not dead, and not upgrading; this frees
+		// their dependencies from model-lifetime/upgrade concerns.
+		migrationFortressName: ifNotUpgrading(ifNotDead(fortress.Manifold())),
```

Comment:
Is it worth combining `ifNotUpgrading(ifNotDead(` into one decorator? I find this a bit hard to understand.